worlds/banjo_tooie/BTClient.py

```diff
@@ -0,0 +1,1248 @@
+import asyncio
+from dataclasses import dataclass
+import hashlib
+import io
+import json
+import os
+import multiprocessing
+import copy
+import pathlib
+import subprocess
+import sys
+import time
+from typing import List, Tuple, Union
+import zipfile
+import bsdiff4
+import atexit
+
+
+# CommonClient import first to trigger ModuleUpdater
+from CommonClient import CommonContext, server_loop, gui_enabled, \
+    ClientCommandProcessor, logger, get_base_parser
+import Utils
+import settings
+from Utils import async_start
+from worlds import network_data_package
+
+SYSTEM_MESSAGE_ID = 0
+
+CONNECTION_TIMING_OUT_STATUS = "Connection timing out. Please restart your emulator, then restart banjoTooie_connector.lua"
+CONNECTION_REFUSED_STATUS = "Connection refused. Please start your emulator and make sure banjoTooie_connector.lua is running"
+CONNECTION_RESET_STATUS = "Connection was reset. Please restart your emulator, then restart banjoTooie_connector.lua"
+CONNECTION_TENTATIVE_STATUS = "Initial Connection Made"
+CONNECTION_CONNECTED_STATUS = "Connected"
+CONNECTION_INITIAL_STATUS = "Connection has not been initiated"
+
+"""
+Payload: lua -> client
+{
+    playerName: string,
+    locations: dict,
+    deathlinkActive: bool,
+    taglinkActive: bool,
+    isDead: bool,
+    isTag: bool,
+    gameComplete: bool
+}
+
+Payload: client -> lua
+{
+    items: list,
+    playerNames: list,
+    triggerDeath: bool,
+    triggerTag: bool,
+    messages: string
+}
+
+Deathlink logic:
+"Dead" is true <-> Banjo is at 0 hp.
+
+deathlink_pending: we need to kill the player
+deathlink_sent_this_death: we interacted with the multiworld on this death, waiting to reset with living link
+
+"""
+
+bt_loc_name_to_id = network_data_package["games"]["Banjo-Tooie"]["location_name_to_id"]
+bt_itm_name_to_id = network_data_package["games"]["Banjo-Tooie"]["item_name_to_id"]
+script_version: int = 5
+version: str = "V4.11.3"
+patch_md5: str = "54efb965e58f1bb15487d411dc8955a3"
+bt_options = settings.get_settings().banjo_tooie_options
```

Comment:
It might be better to import the world to get this data, e.g. `BanjoTooieWorld.settings` or `BanjoTooieWorld.location_name_to_id` instead of `get_settings()` or `network_data_package` (which means importing *all* worlds). Some other worlds' clients seem to this.